controllers/webhook.js

```diff
@@ -0,0 +1,302 @@
+  if (!event) {
+    errorsRaised.push('No event found in the request')
+  } else {
+    if (event === 'Merge Request Hook') {
+      let webhookSecretExpected = null
+      if (staticman) {
+        // Webhook request authentication is NOT supported in v1 of the endpoint.
+        await staticman.getSiteConfig().then((siteConfig) => {
+          webhookSecretExpected = siteConfig.get('gitlabWebhookSecret')
+        })
+      }
+
+      let reqAuthenticated = true
+      if (webhookSecretExpected) {
+        reqAuthenticated = false
+        const webhookSecretSent = req.headers['x-gitlab-token']
+        if (!webhookSecretSent) {
+          // This could be worth logging... unless the endpoint gets hammered with spam.
+          errorsRaised.push('No secret found in the webhook request')
+        } else if (webhookSecretExpected === webhookSecretSent) {
+          /*
+           * Whereas GitHub uses the webhook secret to sign the request body, GitLab does not.
+           * As such, just check that the received secret equals the expected value.
+           */
+          reqAuthenticated = true
+        } else {
+          // This could be worth logging... unless the endpoint gets hammered with spam.
+          errorsRaised.push('Unable to verify authenticity of request')
+        }
+      }
+
+      if (reqAuthenticated) {
+        await _handleMergeRequest(req.params, service, req.body, staticman).catch((errors) => {
+          errorsRaised = errors
+        })
+      }
+    }
+  }
+
+  if (errorsRaised.length > 0) {
+    return Promise.reject(errorsRaised)
+  }
+}
+
+const _calcIsVersion1WhenGitHubAssumed = function (req) {
+  const service = req.params.service
+  const version = req.params.version
+  return (!service && version === '1')
+}
+
+const _verifyGitHubSignature = function (secret, data, signature) {
+  const signedData = 'sha1=' + crypto.createHmac('sha1', secret).update(data).digest('hex')
+  return bufferEq(Buffer.from(signature), Buffer.from(signedData))
+}
+
+const _handleMergeRequest = async function (params, service, data, staticman) {
+  const errors = []
+
+  const ua = config.get('analytics.uaTrackingId')
+    ? require('universal-analytics')(config.get('analytics.uaTrackingId'))
+    : null
+
+  const version = params.version
+  const username = params.username
+  const repository = params.repository
+  const branch = params.branch
+
+  let gitService = null
+  let mergeReqNbr = null
+  if (service === 'github') {
+    gitService = await _buildGitHubService(version, service, username, repository, branch, data)
+    mergeReqNbr = data.number
+  } else if (service === 'gitlab') {
+    gitService = await gitFactory.create('gitlab', {
+      version: version,
+      username: username,
+      repository: repository,
+      branch: branch
+    })
+    mergeReqNbr = data.object_attributes.iid
+  } else {
+    errors.push('Unable to determine service.')
+    return Promise.reject(errors)
+  }
```

Comment:
```suggestion
  const version = params.version
  const username = params.username
  const repository = params.repository
  const branch = params.branch

  let gitService = await _buildGitService(version, service, username, repository, branch, data)
  let mergeReqNbr = getMergeRequestNumber(service);
  
  ------------------------------------------------------------------------------------------

const _buildGitService = function (version, service, username, repository, data) {
  // Required for v1 webhook backwards compatibility
  username = username ? username : data.repository.owner.login
  repository = repository ? repository : data.repository.name
  branch = branch ? branch : data.pull_request.base.ref
 
  const serviceConfig = {
    version,
    username,
    repository,
    branch
  }
  return await gitFactory.create(service, serviceConfig);
}

const _getMergeRequestNumber = function (service, data) {
  return service === 'github' ? data.number : data.object_attributes.iid
}
```

We can do something like above and make a helper function to abstract out the git service creation from webhook handler. This way we don't have to manually add cases to the conditional in the event another service is eventually supported.